### DISCUSSION TITLE
PCA labels

### DISCUSSION THREAD

**Discussion by mw00847** [spectrochempy/spectrochempy#496] [General]
Have used pca.reduce as below,  with a .SPG file.

How can I label the plot identifying which spectrum it came from?

using this creates a list of the spectrum names, but I cant see how to use it.

names=dataset.y.labels[:,1]

------------------

pca = scp.PCA(dataset, centered=True, standardized=True,scaled=True)

S, LT = pca.reduce(n_pc=0.99)

ev_ratio = pca.reduce(n_pc=3)

#_ = pca.screeplot()

_=pca.scoreplot(2,3)

**Comment by mw00847** [comment]
Have used sklearn and plotly to interactively identify the original spectrum along a principal component. As below,

import pandas as pd 
from sklearn.decomposition import PCA
import numpy
import matplotlib.pyplot as plt
import plotly.express as px
from sklearn import decomposition 

#using pandas dataframes for each spectrum 
df2=pd.read_csv('A.csv',usecols=[1])
df3=pd.read_csv('B.csv',usecols=[1])
df4=pd.read_csv('C.csv',usecols=[1])
df5=pd.read_csv('D.csv',usecols=[1])
df6=pd.read_csv('E.csv',usecols=[1])
df7=pd.read_csv('F.csv',usecols=[1])

#concatenate the dataframes and name the spectrum

df=pd.concat([df1,df2,df3,df4,df5,df6,df7])
names =(["A","B","C","D","E","F"])

#normalise the data

df_normalized=(df - df.mean()) / df.std()
pca = PCA(n_components=df.shape[1])
pca.fit(df_normalized)

#pca=decomposition.PCA(n_components=26)
#pca.fit(df_normalized)
X=pca.transform(df_normalized)

loadings = pd.DataFrame(pca.components_.T,
columns=['PC%s' % _ for _ in range(len(df_normalized.columns))],
index=df.columns)

loadings['columns'] = columns

#fig=px.scatter(loadings,x="PC1",y="PC2",color="columns")

fig =px.scatter_3d(loadings,x="PC4",y="PC5",z="PC6",color='columns')

fig.show()

**Reply by fernandezc** [reply]
Hello,
Sorry I didn't get to the original question in time.
I think it should be possible to do the same thing in SpectroChemPy but I need to look into it further. I will come back as soon as possible. I put it as an issue to keep track on it.

**Comment by fernandezc** [comment]
This is now introduced as an option in SpectroChemPy (see https://www.spectrochempy.fr/latest/gallery/auto_examples/analysis/plot_pca_spec.html#sphx-glr-gallery-auto-examples-analysis-plot-pca-spec-py)
